src/core/mesh/qgsmeshlayerinterpolator.cpp

```diff
@@ -131,6 +131,18 @@ QgsRasterBlock *QgsMeshLayerInterpolator::block( int, const QgsRectangle &extent
     int topLim, bottomLim, leftLim, rightLim;
     QgsMeshLayerUtils::boundingBoxToScreenRectangle( mContext.mapToPixel(), mOutputSize, bbox, leftLim, rightLim, topLim, bottomLim );
 
+    double value( 0 ), value1( 0 ), value2( 0 ), value3( 0 );
+    const int faceIdx = mTriangularMesh.trianglesToNativeFaces()[triangleIndex];
```

Comment:
`faceIdx  = mTriangularMesh.trianglesToNativeFaces()[triangleIndex];`  could be in else branch on line 144 since we do not need it for data on vertices